clang/test/CXX/drs/dr0xx.cpp

```diff
@@ -1414,7 +1414,7 @@ namespace dr96 { // dr96: no
     // FIXME: This is ill-formed, because 'f' is not a template-id and does not
```

Comment:
I guess this FIXME is superseded by P1787R6, the same way 2010 resolution is.

Reply:
I'm not sure what direction of "superseded" you mean here, but (except for the separable alias-template bit) the comment is simply correct (since CWG96; before that any template would do).  All that P1787R6 did here was correct the strange use of "name" to refer to (among other things) a *template-id* and deprecate the `T::template NestedTypeTemplate` case.

Reply:
@opensdh Thank you for the clarification. I guess I read [N1528](https://wg21.link/n1528)
Would it be a correct statement to make that you deprecated the following case described in that paper?
> Consequently, the recommendation of this paper is: If the template keyword is followed by a name that does not have a template-argument-list, the name must refer to a class template and be used in a context in which a class template is valid, otherwise the program is ill-formed.

Reply:
Yes, P1787R6 deprecated that use case; you're supposed to just not use `template` there.  This is consistent with the _recommendation_ in N1528, but of course we now reject the premise that `template` is _needed_ for a template template argument.  The reasoning, if it helps, is that compilers already have to deal with ambiguity there:
```cpp
template<auto> void f();                                // #1
template<class> void f(int=0);                          // #2
template<template<class> class> void f(void*=nullptr);  // #3

template<class T> void g() {
  // in C++20, [temp.res]/6 and [temp.arg.template]/1 contradict each other here:
  f<T::Q>();             // could be #1 or #3
}
template<class T> struct X {
  // OK per [temp.local]/1
  void h() {f<X>(T());}  // could be #2 or #3
  void i() {T::template R<X>();}  // could be anything
}
```
Accordingly, [CWG1478](https://cplusplus.github.io/CWG/issues/1478.html)'s question was answered in the negative, and `template` (is to be) restricted to the case where it influences the interpretation of a `<`.